functions/assert_metadata_os.pp

```diff
@@ -0,0 +1,57 @@
+# Fails a compile if the client system is not compatible with the module's
```

Comment:
@jeefberkey This replaces the logic that used to be all over `params.pp` where you had a catch statement that failed on unsupported Operating Systems. Now, instead of a big block of logic, you can just call assert and have it take care of it for you.